src/NuGet.Core/NuGet.Packaging/Signing/Authoring/AuthorSignPackageRequest.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Security.Cryptography.X509Certificates;
+using NuGet.Common;
+
+namespace NuGet.Packaging.Signing
+{
+    public sealed class AuthorSignPackageRequest : SignPackageRequest
+    {
+        /// <summary>
+        /// Instantiates a new instance of the <see cref="AuthorSignPackageRequest" /> class.
+        /// </summary>
+        /// <param name="certificate">The signing certificate.</param>
+        /// <param name="signatureHashAlgorithm">The signature hash algorithm.</param>
+        /// <exception cref="ArgumentNullException">Thrown if <paramref name="certificate" />
+        /// is <c>null</c>.</exception>
+        /// <exception cref="ArgumentException">Thrown if <paramref name="signatureHashAlgorithm" />
+        /// is invalid.</exception>
+        public AuthorSignPackageRequest(
+            X509Certificate2 certificate,
+            HashAlgorithmName signatureHashAlgorithm)
```

Comment:
why does author signature have the same hash algorithm for signature and timestamp but repository signatures can have different algorithms?

Reply:
Author signatures can have different hash algorithms too, but that is a [different constructor overload](https://github.com/NuGet/NuGet.Client/pull/1968/files/e733786817e96e9eab0ca1e5186ac8a4f80de360..f1f77d6314f3eb5aca8f2f01a0fe1bab74dba94a#diff-0ca9b9798c734d2097235a2ae782679eR45).

This constructor is a convenience overload and is not absolutely necessary.  (It pre-exists my current change.)

Reply:
Awesome! Should we change the name of the parameter from `signatureHashAlgorithm` to something more general like `hashAlgorithm`?